d3a_api_client/commands.py

```diff
@@ -13,20 +13,40 @@ def __init__(self, ):
     def buffer_length(self):
         return len(self._commands_buffer)
 
-    def offer_energy(self, area_uuid, energy, price):
-        return self._add_to_buffer(area_uuid, Commands.OFFER, {"energy": energy, "price": price})
+    def offer_energy(
+            self, area_uuid: str, energy: float, price: float, replace_existing: bool = True):
 
-    def offer_energy_rate(self, area_uuid, energy, rate):
-        return self._add_to_buffer(area_uuid, Commands.OFFER, {"energy": energy, "price": rate * energy})
+        return self._add_to_buffer(
+            area_uuid,
+            Commands.OFFER,
+            {'energy': energy, 'price': price, 'replace_existing': replace_existing})
+
+    def offer_energy_rate(
+            self, area_uuid: str, energy: float, rate: float, replace_existing: bool = True):
+
+        return self._add_to_buffer(
+            area_uuid,
+            Commands.OFFER,
+            {'energy': energy, 'price': rate * energy, 'replace_existing': replace_existing})
 
     def update_offer(self, area_uuid, energy, price):
         return self._add_to_buffer(area_uuid, "update_offer", {"energy": energy, "price": price})
 
-    def bid_energy(self, area_uuid, energy, price):
-        return self._add_to_buffer(area_uuid, Commands.BID, {"energy": energy, "price": price})
+    def bid_energy(
+            self, area_uuid: str, energy: float, price: float, replace_existing: bool = True):
+
+        return self._add_to_buffer(
+            area_uuid,
+            Commands.BID,
+            {'energy': energy, 'price': price, 'replace_existing': replace_existing})
```

Comment:
Just a minor question. Is this refactoring following a best practice layout?
When I would have implemented it, adding the `replace_existing` parameter would have added one line to the `bid_energy`. Now the function has 7 lines and IMO the readability did not improve IMO.
Happy to learn how to do it correctly.

Reply:
Do you have an example of how you would have implemented it?
Adding type hints already makes the signature longer than 100 characters. Because of that, I added a new line after `def bid_energy(`.
The same applies to the parameters for `self._add_to_buffer`: they were longer than 100 characters, so I had to break them somehow. Since the last parameter (the dict) takes almost one line, I decided to allocate one single line to each parameter because I thought it looked clearer. 
There are lots of ways to break continuation lines though; I usually use [hanging indents](https://www.python.org/dev/peps/pep-0008/#indentation) when I can, and let use my personal taste when I'm in doubt.

Reply:
Having read the link to the corresponding paragraph in PEP8 you posted, I agree with your way of implementing (my proposal would have been one of the solutions under the `# Wrong:` part :-D)